openassessment/xblock/grade_mixin.py

```diff
@@ -130,6 +130,49 @@ def render_grade_complete(self, workflow):
         # when all the criteria in the rubric are feedback-only (no options).
         score = workflow['score']
 
+        first_sentence = "You have successfully completed this problem and received a {}/{}.".format(
+            score["points_earned"], score["points_possible"]
+        )
+        second_sentence = ""
+
+        is_self_assessment = "self-assessment" in assessment_steps
+        is_peer_assessment = "peer-assessment" in assessment_steps
+        is_staff_assessment = "staff-assessment" in assessment_steps
+
+        if is_self_assessment and not (is_staff_assessment or is_peer_assessment):
+            second_sentence = "The grade for this problem is determined by your Self Assessment."
+
+        elif is_staff_assessment and not (is_self_assessment or is_peer_assessment):
+            second_sentence = "The grade for this problem is determined by your Staff Grade."
+
+        elif is_peer_assessment and not (is_staff_assessment or is_self_assessment):
+            # TODO: Some additions to this message will be necessary to this message
+            #       based on E and H milestones.
+            second_sentence = (
+                "The grade for this problem is determined by the average overall "
+                "score of your Peer Assessments."
+            )
+
+        elif is_self_assessment and is_peer_assessment and not is_staff_assessment:
+            # TODO: Sentence 2 not mentioned yet by the reporter. The following would be replaced
+            #       once I get the actual sentence
+            second_sentence = "This is self + peer assessment and Sentence 2 isn't decided yet"
+
+        elif is_self_assessment and is_staff_assessment and not is_peer_assessment:
+            # TODO: Sentence 2 not mentioned yet by the reporter. The following would be replaced
+            #       once I get the actual sentence
+            second_sentence = "This is self + staff assessment and Sentence 2 isn't decided yet"
+
+        elif is_staff_assessment and is_peer_assessment and not is_self_assessment:
+            # TODO: Sentence 2 not mentioned yet by the reporter. The following would be replaced
+            #       once I get the actual sentence
+            second_sentence = "This is staff + peer assessment and Sentence 2 isn't decided yet"
+
+        elif is_self_assessment and is_staff_assessment and is_peer_assessment:
+            second_sentence = "The grade for this problem is determined by your Staff Grade."
+
+        score["explanation"] = first_sentence + " " + second_sentence
```

Comment:
`score` is directly returned from the Workflow API, don't modify it. Instead add a new variable to the context bellow.